src/plugins/parser/tip.js

```diff
@@ -1,34 +1,133 @@
-module.exports = function mention() {
+let redact;
+
+const removeIndentation = require('remark-parse/lib/util/remove-indentation');
+
+const RE = /^!!! ?([\w-]+)(?: "(.*?)")?(?: <(.*?)>)?\n/
+
+module.exports = function tip() {
   const Parser = this.Parser;
-  const tokenizers = Parser.prototype.inlineTokenizers;
-  const methods = Parser.prototype.inlineMethods;
+  const tokenizers = Parser.prototype.blockTokenizers;
+  const methods = Parser.prototype.blockMethods;
+  const restorationMethods = Parser.prototype.restorationMethods;
+
+  restorationMethods.tip = function (add, nodes, content, children) {
+    let value = `!!!${nodes.open.tipType}`;
+    if (content) {
+      value += ` "${content}"`;
+    }
+    if (nodes.open.id) {
+      value += ` <${nodes.open.id}>`;
+    }
+    return add({
+      type: 'paragraph',
+      children: [{
+        type: 'rawtext',
+        value: value + "\n"
+      }, {
+        type: 'indent',
+        children
+      }]
+    });
+  }
+
+  redact = Parser.prototype.options.redact;
 
-  /* Add an inline tokenizer (defined in the following example). */
-  tokenizers.mention = tokenizeMention;
+  tokenizers.tip = tokenizeTip;
 
-  /* Run it just before `text`. */
-  methods.splice(methods.indexOf('text'), 0, 'mention');
+  /* Run it just before `paragraph`. */
+  methods.splice(methods.indexOf('paragraph'), 0, 'tip');
 }
 
-tokenizeMention.notInLink = true;
-tokenizeMention.locator = locateMention;
+function tokenizeTip(eat, value, silent) {
+  const match = RE.exec(value);
+  if (!match) {
+    return;
+  }
 
-function tokenizeMention(eat, value, silent) {
-  const match = /^@(\w+)/.exec(value);
+  if (silent) {
+    return true;
+  }
 
-  if (match) {
-    if (silent) {
-      return true;
+  // find the indented block that represents the content of the tip. Blocks can
+  // be indented by either four spaces or a tab character
```

Comment:
might be helpful to add: `// Blocks can also contain blank lines separating multiple indented paragraphs.`